### ISSUE TITLE
BayesianModel __eq__ method

### ISSUE THREAD

**Issue by AlexGoico** [pgmpy/pgmpy#897]
I noticed other classes such as DiscreteFactor have equality but BayesianModel does not. Perhaps something as simple as the following will do?
```
def __eq__(self, other):
    if isinstance(other, self.__class__):
        return self.__dict__ == other.__dict__
    else:
        return False
```

**Comment by ankurankan** [comment]
@AlexGoico Yes, I think it should work. If you would like, you are most welcome to send a pull request for this. And I think we can have something like this for other model classes as well.
